### DISCUSSION TITLE
Integrate Mullvad-Check Tool as docker environment with gluetun

### DISCUSSION THREAD

**Discussion by isudoi** [qdm12/gluetun#446] [Ideas]
_**### RFE - Question**_

**>>>> Summary:**
I just deployed a new setup using Raspberry Pi4 (Did [step 2](https://github.com/alpinelinux/docker-alpine/issues/135#issuecomment-812287338) in ReadMe) ;)
And port forwarding feature with Mullvad
Things working great

**>>>> Maybe this is RFE:**
Is it possible to add the following to your list of [HTTP API](https://github.com/qdm12/gluetun/wiki/HTTP-Control-server)
https://ipv4.am.i.mullvad.net/port/54823
This is from: https://mullvad.net/en/check/

For example, with Mullvad, where it uses the same env in docker compose
`      - FIREWALL_VPN_INPUT_PORTS=54823`
And then returns the status result

**>>>> Why I think this is would be good idea?**
With that, I can run a PowerShell/Python script to send me Telegram/Email on-demand to retrieve both public IP address and the port (I know it's static with Mullvad)

Also, it's convenient to have that feature for a faster access to resources

Cheers

**Comment by qdm12** [comment]
It looks like https://ipv4.am.i.mullvad.net/port/54823 only checks if the port is reachable at your IP. For example without VPN accessing https://ipv4.am.i.mullvad.net/port/443 gives me true with my public IP address.

You can retrieve the public IP Address and the forwarded port from the API already, in a Mullvad neutral way.

However, I can add something to try reaching the forwarded port, probably return an extra field (true/false) when you access the forwarded port in the http control server, that could be cool.

**Reply by isudoi** [reply]
In the meantime, I have created a PowerShell 7 script on always-on Windows 10 MiniPC, that makes a call to the Gluetun HTTP Control Server and retrieve the current VPN IP and adds the port into a text format along with the IP

`**Using:** `https://github.com/qdm12/gluetun/wiki/HTTP-Control-server

Example:
https://1.1.1.1:3214

And then it sends via PowerShell 7 to Telegram bot chat
`**Using:** `https://github.com/techthoughts2/PoshGram

It's a workaround for RFE:
https://github.com/qdm12/gluetun/issues/397

With all of that, whenever I need to access my resources remotely over the Port Forwarding VPN, I just hit remotely request for the schedule task to run and it sends me the complete URL (http://IP:Port)
`**Using:** `https://intercom.help/pulseway/en/articles/1514296-enable-scheduled-task-monitoring


I know a lot of tools being used, however, the purpose was served 💯
